source/common/upstream/health_checker_impl.cc

```diff
@@ -285,7 +286,7 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onInterval() {
   request_in_flight_ = true;
 
   const auto request_headers = Http::createHeaderMap<Http::RequestHeaderMapImpl>(
```

Comment:
There is a potential need to make the request body configurable for POST, PUT, and PATCH requests. However, we probably don't want to over-skill it because this is only for health check.

Reply:
#2 is fine at current stage. We can add a comment that request body is not supported right now.